python_modules/libraries/dagster-snowflake-pyspark/dagster_snowflake_pyspark/snowflake_pyspark_type_handler.py

```diff
@@ -0,0 +1,163 @@
+from typing import Mapping
+
+import dagster._check as check
+from dagster import InputContext, MetadataValue, OutputContext, TableColumn, TableSchema
+from dagster._core.definitions.metadata import RawMetadataValue
+from dagster._core.storage.db_io_manager import DbTypeHandler, TableSlice
+from dagster_snowflake import build_snowflake_io_manager
+from pyspark.sql import DataFrame, SparkSession
+
+SNOWFLAKE_CONNECTOR = "net.snowflake.spark.snowflake"
+
+
+def _get_sf_options(config, table_slice):
+    check.invariant(
+        config.get("warehouse", None) is not None,
+        "Missing config: Warehouse is required when using PySpark with the Snowflake I/O manager.",
+    )
+
+    conf = {
+        "sfURL": f"{config['account']}.snowflakecomputing.com",
+        "sfUser": config["user"],
+        "sfPassword": config["password"],
+        "sfDatabase": config["database"],
+        "sfSchema": table_slice.schema,
+        "sfWarehouse": config["warehouse"],
+        "dbtable": table_slice.table,
+    }
+
+    return conf
+
+
+class SnowflakePySparkTypeHandler(DbTypeHandler[DataFrame]):
+    """
+    Plugin for the Snowflake I/O Manager that can store and load PySpark DataFrames as Snowflake tables.
+
+    Examples:
+        .. code-block:: python
+
+            from dagster_snowflake import build_snowflake_io_manager
+            from dagster_snowflake_pyspark import SnowflakePySparkTypeHandler
+            from pyspark.sql import DataFrame
+            from dagster import Definitions
+
+            snowflake_io_manager = build_snowflake_io_manager([SnowflakePySparkTypeHandler()])
+
+            @job(resource_defs={'io_manager': snowflake_io_manager})
+            def my_job():
+                ...
+
+
+            # OR
+
+            @asset
+            def my_asset() -> DataFrame:
+                ...
+
+            defs = Definitions(
+                assets=[my_asset],
+                resources={
+                    "io_manager": snowflake_io_manager.configured(...)
+                }
+            )
+    """
+
+    def handle_output(
+        self, context: OutputContext, table_slice: TableSlice, obj: DataFrame
+    ) -> Mapping[str, RawMetadataValue]:
+        options = _get_sf_options(context.resource_config, table_slice)
+
+        with_uppercase_cols = obj.toDF(*[c.upper() for c in obj.columns])
+
+        with_uppercase_cols.write.format(SNOWFLAKE_CONNECTOR).options(**options).mode(
+            "append"
+        ).save()
+
+        return {
+            "row_count": obj.count(),
```

Comment:
`count` can be a very expensive operation in Spark, because it basically triggers the entire computation of dataframe, so I think we should avoid invoking it here

Reply:
good to know. do you know of an alternative? should we just not have this metadata for spark dfs?

Reply:
Not that I'm aware of, so yeah, I think it's best to just omit this.